durabletask/worker.py

```diff
@@ -117,7 +129,12 @@ def add_activity(self, fn: task.Activity) -> str:
 
     def start(self):
         """Starts the worker on a background thread and begins listening for work items."""
-        channel = shared.get_grpc_channel(self._host_address, self._metadata, self._secure_channel)
+        
+        if self._metadata:
+            interceptors = [DefaultClientInterceptorImpl(self._metadata)]  
+        else:
+            interceptors =  None
```

Comment:
We already initialized the interceptors list as part of `self._interceptors` in the constructor, so it seems like we shouldn't be doing this here, and below we should be passing in `self._interceptors` to the `shared.get_grpc_channel` call.